rosidl_generator_cpp/resource/msg__struct.hpp.em

```diff
@@ -100,11 +102,7 @@ def generate_default_string(membset):
                 strlist.append('this->%s.resize(%d);' % (member.name, member.num_prealloc))
             if isinstance(member.default_value, list):
                 if all(v == member.default_value[0] for v in member.default_value):
-                    # Specifying type for std::fill because of MSVC 14.12 warning about casting 'const int' to smaller types (C4244)
-                    # For more info, see https://github.com/ros2/rosidl/issues/309
-                    # TODO(jacobperron): Investigate reason for build warnings on Windows
-                    # TODO(jacobperron): Write test case for this path of execution
-                    strlist.append('std::fill<typename %s::iterator, %s>(this->%s.begin(), this->%s.end(), %s);' % (msg_type_to_cpp(member.type), msg_type_only_to_cpp(member.type), member.name, member.name, member.default_value[0]))
+                    strlist.append('std::fill(this->%s.begin(), this->%s.end(), %s);' % (member.name, member.name, member.default_value[0]))
```

Comment:
@dirk-thomas I'm curious why the previous comment no longer applies.

Reply:
I don't see any warning on Windows for this line. Also "Write test case for this path of execution" applies to literally every line of code so I don't think it should be preserved.